src/app/services/latis/latis.service.ts

```diff
@@ -1,14 +1,25 @@
 import { HttpClient } from '@angular/common/http';
 import { Injectable } from '@angular/core';
 import * as moment from 'moment';
+import { BehaviorSubject } from 'rxjs';
 import { environment } from 'src/environments/environment';
 
 @Injectable({
     providedIn: 'root'
 })
 export class LatisService {
+    mostRecentAp: BehaviorSubject<number> = new BehaviorSubject(undefined);
 
-    constructor(private http: HttpClient) {}
+    constructor(private http: HttpClient) {
+        // get the most recent ap value on app load
+        this.http.get(`${environment.latisSwp}ap.jsond?last()`).subscribe( (response: any) => {
```

Comment:
Are we now making two calls to get ap?
1. To just get the most recent range on page load
2. To get the data that the user requests

Reply:
Yes, unfortunately. The first call happens on app load. I don't know how else to find the latest available ap data. They don't seem to update it on a regular cadence.

Reply:
On app load can you make a single call to `.last(n) or something like that to get all of the values you need at once? Then, you could just populate the ap data from that single call, set your date/time range, and then make your F10.7 call for the appropriate date.

Then only make further calls when anything else changes. Maybe this means a separate call order in `on_init()` or similar?

That might be overcomplicating things too.

Reply:
Maybe we can chain latis operations? Right now, `last()` does not take any arguments. Maybe `takeRight(n)` would do what we want? I'm not sure if we would need a date for `takeRight(n)`. I'll look into that. Documentation is here: https://lasp.colorado.edu/lisird4/about/latis#operation. Good idea.

Reply:
okay, so, looking at it more deeply, we now make three (3) calls to latis for ap: last(), daily Ap (somewhere from 1 to 8 responses for a date), and the 20 values before the date.

It is possible (I think) to chain latis operations. 

However, the way it is set up, it only does the request to last() once, on app load. That first request bootstraps all the other requests, and each time the date is changed, the other 2 requests are performed to update the values. Doing something different for that first request would be weird. 

I hope that makes sense. If you can think of a better, more concise way to structure it, let me know, but I think, while this might not be the minimum number of requests possible, it makes only one more request than before (once, on app load only), is easier to understand, and keeps each action isolated.

Reply:
That works for me.

I thought that what you said above about `takeRight(n)` could replace the `last()` call, and you could just look at what the most recent value was in that `takeRight(n)` call to update all of your times. But, if that operation returns "null" values too then that wouldn't work anyways. I was more curious than anything, let's not overcomplicate it like you say.